---
title: "RNAseq processing"
author: "Lucas F Maciel"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

As requested by the task, the sequencing reads from the SRR11647 run were downloaded in the FASTQ format and further processed. Below is the report explaining all steps.

### **1** FASTQ sample download

The download was performed using the SRA Toolkit in the following steps:

**1.1** The SRA file was downloaded and its content integrity was verified. 

```{bash,eval=F, echo=T}
#Create directories
mkdir fastq fastq/raw
# Download SRA files
cd fastq
prefetch SRR1164787
#Check SRA integrity
vdb-validate SRR1164787/
```

**1.2** The FASTQ files were downloaded using fasterq-dump and zipped 

```{bash,eval=F, echo=T}
fastq-dump -O fastq/raw split-files SRR1164787
gzip raw/* 
cd ..
```

## **2** FASTQs quality control

**2.1** The next step was checking the quality of the obtained reads. In order to do that FASTQC was used.

```{bash,eval=F, echo=T}
mkdir fastqc/ fasted/raw
fastqc -o fastqc/raw/ fastq/raw/*
```

```{r,warning=F,message=FALSE,fig.width=8,fig.height=8}
library(fastqcr)
library(cowplot)
library(kableExtra)
qc <- qc_aggregate("fastqc/")

kable(qc)

qc1 <- qc_read("fastqc/SRR1164787_1_fastqc.zip")
p1 <- qc_plot(qc1, "Per base sequence quality")
p2 <- qc_plot(qc1, "Adapter content")

qc2 <- qc_read("fastqc/SRR1164787_2_fastqc.zip")
p3 <- qc_plot(qc2, "Per base sequence quality")
p4 <- qc_plot(qc2, "Adapter content")


cowplot::plot_grid(p1,p2,p3,p4,ncol=2)
```


**2.2** Based on the tables and plots above we can observe that the reads had good quality, however, there are some unpaired reads as the numbers of reads on R1 and R2 differ. I believe this happens because the files originally uploaded to SRA were BAM files, and not the raw reads. 
Thus, trimmomatic was used just to get rid of unpaired reads. 

```{bash,eval=F, echo=T}
#Run trimmomatic to remove unpaired reads
cd fastq
mkdir trimmed
java -jar trimmomatic-0.39.jar PE -threads 4 raw/SRR1164787_1.fastq.gz raw/SRR1164787_2.fastq.gz  trimmed/SRR1164787_1.fastq.gz trimmed/SRR1164787.unpaired1.fastq.gz trimmed/SRR1164787_2.fastq.gz trimmed/SRR1164787.unpaired2.fastq.gz MINLEN:40
```

## **3** Reads mapping against human genome

The next step was to map the reads against the human genome. According to the metadata available at GEO, in 2014 they used Tophat and Bowtie to perform this mapping step. I have decided to use STAR, which is a much faster and up-to-date method, being widely used in RNAseq workflows, including by the nf-core RNA-seq pipeline (https://nf-co.re/rnaseq/3.14.0/).

**3.1** The provided human genome and transcriptome annotation were downloaded.

```{bash,eval=F, echo=T}
mkdir  annotations
cd annotations
#Download GTF
curl https://ftp.ensembl.org/pub/release-113/gtf/homo_sapiens/Homo_sapiens.GRCh38.113.gtf.gz --output Homo_sapiens.GRCh38.113.gtf.gz
#Download FASTA file
curl https://ftp.ensembl.org/pub/release-113/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz --output Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
#Check download integrity
sum * 
44004 62639 Homo_sapiens.GRCh38.113.gtf.gz
22450 861294 Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
#Unzip 
gunzip *
```

**3.2** The next step was to build a reference index

```{bash,eval=F, echo=T}
#Generate STAR reference index
mkdir star_index_dir
STAR --runMode genomeGenerate --runThreadN 10 --genomeDir star_index_dir \
 --genomeFastaFiles Homo_sapiens.GRCh38.dna.primary_assembly.fa \
 --sjdbGTFfile Homo_sapiens.GRCh38.113.gtf --sjdbOverhang 49
```

**3.3** And finally the reads could be mapped to the genome.

```{bash,eval=F, echo=T}
STAR --runThreadN 10 --genomeDir /staging/leuven/stg_00054/ariel/lucas/annotations/star_index_dir \  
  --readFilesIn /staging/leuven/stg_00054/ariel/lucas/fastq/trimmed/SRR1164787_1.fastq.gz \  
  /staging/leuven/stg_00054/ariel/lucas/fastq/trimmed/SRR1164787_2.fastq.gz \
  --readFilesCommand zcat \  
  --outFileNamePrefix SRR1164787_ --quantMode TranscriptomeSAM \ 
  --outSAMtype BAM SortedByCoordinate --limitBAMsortRAM 112924475913
```

STAR generates a final metric report. According to the information available on GEO, a total of 63% of all reads were properly mapped, however, this metric is not related to only the sample processed in this task. In this particular task, around 80% of reads were mapped (37% uniquely mapped) by STAR.

```{bash}
cat SRR1164787_Log.final.out
```
## **4** Gene quantification

**4.1** The final step was the quantification of expression based on the mapped reads. The BAM file generated by STAR was used as input to RSEM.

```{bash,eval=F, echo=T}
#RSEM also requires an index to be built, so this has to be done first.
mkdir rsem_index
rsem-prepare-reference --num-threads 10 --star --gtf Homo_sapiens.GRCh38.113.gtf Homo_sapiens.GRCh38.dna.primary_assembly.fa rsem_index/rsem_reference 

#Now calculate expression
rsem-calculate-expression --num-threads 10 --paired-end --alignments \ SRR1164787_Aligned.toTranscriptome.out.bam rsem_index/rsem_reference rsem_SRR1164787_
```

**4.2** This finally allowed us to obtain the expression levels of CTNNB1 (ENSG00000168036), which was the last goal of this analysis.

```{r}
rsem <- read.table("rsem_SRR1164787_.genes.results",sep = "\t",header = T)
rsem[rsem$gene_id == "ENSG00000168036",c(1,3:7)]
```

